src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt

```diff
@@ -32,31 +33,48 @@ class ImportNameIntention : RsElementBaseIntentionAction<ImportNameIntention.Con
         if (basePath.reference.resolve() != null) return null
         val pathMod = path.containingMod
         val pathSuperMods = HashSet(pathMod.superMods)
-        // TODO: support reexports
-        val candidates = RsNamedElementIndex.findElementsByName(project, basePath.referenceName)
+
+        val explicitItems = RsNamedElementIndex.findElementsByName(project, basePath.referenceName)
             .asSequence()
             .filterIsInstance<RsQualifiedNamedElement>()
-            .filter { basePath != path || !(it is RsMod || it is RsModDeclItem || it.parent is RsMembers) }
-            .mapNotNull { item -> (item as? RsVisible)?.canBeImported(pathSuperMods)?.let { ImportCandidate(item, it) } }
+            .map { ImportItem.ExplicitItem(it) }
+
+        val reexportedItems = RsReexportIndex.findReexportsByName(project, basePath.referenceName)
+            .asSequence()
+            .mapNotNull {
+                val item = it.path?.reference?.resolve() as? RsQualifiedNamedElement ?: return@mapNotNull null
+                ImportItem.ReexportedItem(it, item)
+            }
+
+        val candidates = (explicitItems + reexportedItems)
```

Comment:
I am a bit worried that a single item can be both inside explicit and reexported items (if it is reexported under it's original name). I don't know a way to avoid it though :(

Reply:
It's next step of auto import improvement
I think we can use some heuristic rules, for example, if there are items from `std` and from other crates, exclude items from other crates. Or drop `mod::sub_mod::item` if there is reexport of this item in `mod`.

Reply:
At least `PyCharm` uses the last rule